### PR TITLE
CIV-6073 : Unit test State Flow Engine - Unspec create claim journey (1V1 & MP)

### PR DESCRIPTION
**Pull request by bhagyashreesharma90**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-6073

### Change description ###
Added test for create claim journey for unspec claim
Added 2 new classes StateFlowEngineUnspecTest and CaseDataBuilderUnspec for this

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineUnspecTest.java [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+            assertThat(stateFlow.getFlags()).hasSize(4).contains(
+                entry(FlowFlag.SPEC_RPA_CONTINUOUS_FEED.name(), false),
+                entry(FlowFlag.NOTICE_OF_CHANGE.name(), false),
+                entry(FlowFlag.GENERAL_APPLICATION_ENABLED.name(), false),
+                entry("RPA_CONTINUOUS_FEED", true)
+            );
+        }
+
+        /**
+         * case 13 : 1V2 both represented unregistered defendant.
```
Comment (m-meulendijks-v1) [inline src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineUnspecTest.java:403]: Did you not have to cover AC 14, or did that get covered somewhere else anyway?

Comment (m-meulendijks-v1) [reply src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineUnspecTest.java:403]: Ah, right you are! :)

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by drummondjm — DISMISSED** [review]

**Review by drummondjm — DISMISSED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineUnspecTest.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        assertThat(stateFlow.getFlags()).contains(
+            entry(FlowFlag.ONE_RESPONDENT_REPRESENTATIVE.name(), true)
+        );
+        assertThat(stateFlow.getFlags()).hasSize(6);    // bonus: if this fails, a flag was added/removed but tests were not updated
+    }
+
+    @ParameterizedTest(name = "{index}: The state flow flags ONE_RESPONDENT_REPRESENTATIVE " +
+        "and TWO_RESPONDENT_REPRESENTATIVES are present and set to the correct values (for appropriate cases)")
+    @MethodSource("caseDataStreamTwoRespondentRepresentatives")
+    void shouldReturnClaimSubmitted_whenCaseDataAtStateClaimSubmitted1v2DifferentSolicitor(CaseData caseData) {
```
Comment (akriti-hmcts) [inline src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineUnspecTest.java:149]: Can you update this name shouldReturnClaimSubmitted_whenCaseDataAtStateClaimSubmitted1v2DifferentSolicitor
to say - whenCaseDataAtStateClaimSubmitted1v2DifferentSolicitorBothRepresented

**Review by akriti-hmcts — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineUnspecTest.java (hunk 2 above)
Comment (bhagyashreesharma90) [reply src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineUnspecTest.java]: Done

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by drummondjm — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Pull request merged**